### PR TITLE
fix(cachable-nft): support auto cache

### PR DESCRIPTION
**Pull request by kingster-will**
## Description
Implement automatic caching support that activates when the block base fee exceeds a specified threshold.

### CODE REVIEW

File: contracts/story-nft/CachableNFT.sol [hunk 1]
```diff
@@ -12,23 +12,37 @@ import { EnumerableMap } from "@openzeppelin/contracts/utils/structs/EnumerableM
...
+        Auto
+    }
+
     /// @dev Storage structure for the CacheableNFT
     /// @custom:storage-location erc7201:story-protocol-periphery.CacheableNFT
     struct CacheableNFTStorage {
         // tokenId => ipId
         EnumerableMap.UintToAddressMap cache;
-        bool cacheMode;
+        bool DEPRECATED_cacheMode;
```
Comment (sebsadface) [inline contracts/story-nft/CachableNFT.sol:27]: Why can't we just remove this`bool DEPRECATED_cacheMode;`

File: contracts/story-nft/CachableNFT.sol [hunk 2]
```diff
@@ -12,23 +12,37 @@ import { EnumerableMap } from "@openzeppelin/contracts/utils/structs/EnumerableM
 import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 abstract contract CachableNFT is OwnableUpgradeable {
     using EnumerableMap for EnumerableMap.UintToAddressMap;
+
+    enum CacheMode {
```
[the remark this answers is not in the document]
Comment (kingster-will) [reply contracts/story-nft/CachableNFT.sol:19]: Added comments.

File: contracts/story-nft/CachableNFT.sol [hunk 3]
```diff
@@ -12,23 +12,37 @@ import { EnumerableMap } from "@openzeppelin/contracts/utils/structs/EnumerableM
...
-    /// @param useCache The new cache mode, true for cache mode, false for passthrough mode.
-    function setCacheMode(bool useCache) external onlyOwner {
+    /// @param mode The new cache mode.
+    function setCacheMode(CacheMode mode) external onlyOwner {
         CacheableNFTStorage storage $ = _getCacheableNFTStorage();
-        $.cacheMode = useCache;
+        $.mode = mode;
+    }
+
+    function setCacheModeAutoThreshold(uint256 threshold) external onlyOwner {
```
[the remark this answers is not in the document]
Comment (kingster-will) [reply contracts/story-nft/CachableNFT.sol:48]: added comments.

File: contracts/story-nft/CachableNFT.sol (hunk 1 above)
Comment (kingster-will) [reply contracts/story-nft/CachableNFT.sol:30]: It will break upgradeability.

**Review by sebsadface — APPROVED** [review]
LGTM :)

**Pull request merged**